sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubsException.cs

```diff
@@ -208,7 +208,10 @@ public enum FailureReason
             ServiceTimeout,
 
             /// <summary>There was a general communications error encountered when interacting with the Azure Event Hubs service.</summary>
-            ServiceCommunicationProblem
+            ServiceCommunicationProblem,
+
+            /// <summary>The exception was a result of a failed attempt to claim partition ownership.</summary>
+            ClaimPartitionFailed,
```

Comment:
@jsquire It looks like this change triggered a generated code error as well. Does still need to go through extra review?

Reply:
Technically, yes.   If we're only using it internally, we probably want to try to hide a flag in the `Data` bag first... but if we think this is something we want to surface to developers and that they'll need to take action on, I can drive the conversations.   

Since it's a new member to an embedded specialized type, it wouldn't be difficult so long as we have a couple of concrete scenarios to pitch.